src/qt/receivecoinsdialog.cpp

```diff
@@ -99,6 +99,9 @@ void ReceiveCoinsDialog::setModel(WalletModel *_model)
         } else {
             ui->useBech32->setCheckState(Qt::Unchecked);
         }
+
+        // eventually disable the main receive button if private key operations are disabled
```

Comment:
How does this interact with multiwallet? Loading/unloading wallets at run time?

Reply:
I still think multiwallet should use one window per wallet rather than a dropdown inside a single window. Pending that, receive screen should at least be visible if any of the wallets aren't read-only. The receive button itself would have to be updated when the user switches between wallets.

Another question is: provided there's enough warnings (including about scams), why shouldn't a user be able to receive money on a watch-only address?

Reply:
@Sjors comment is orthogonal.

@jnewbery: this seems to work perfectly fine with mtulitwallet. Not sure about unloading... but I guess unloading does also call `setModel()`... or can you be more specific why this should be a problem?

Reply:
Yes, you're right that this works fine with multiwallet.

It would be nice if there was help text saying that the 'receive' button is disabled because the wallet has private keys disabled (and so can't generate new receiving addresses). However, that could be added in a separate PR.